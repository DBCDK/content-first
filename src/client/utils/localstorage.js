const supported = window && window.localStorage;

export const setItem = (key, value, version) => {
  if (supported) {
    const entry = {
      // modified: (new Date()).getTime(),
      version,
      value
    };
    localStorage.setItem(key, JSON.stringify(entry));
  }
};

export const getItem = (key, version, defaultValue) => {
  if (!supported) {
    return defaultValue;
  }
  const jsonString = localStorage.getItem(key);
  if (!jsonString) {
    return defaultValue;
  }
  try {
    const parsed = JSON.parse(jsonString);
    if (version !== parsed.version) {
      return defaultValue;
    }
    if (!parsed.value) {
      return defaultValue;
    }
    return parsed.value;
  }
  catch (e) {
    console.log(`Could not parse localstorage item with key=${key}`); // eslint-disable-line
    return defaultValue;
  }
};
