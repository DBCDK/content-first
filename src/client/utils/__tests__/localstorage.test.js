import {getItem, setItem} from '../localstorage';

describe('localstorage', () => {

  test('retrieves stored entry, when version matches', () => {
    setItem('somekey', {
      foo: 'bar'
    }, 1);

    expect(getItem('somekey', 1, {some: 'fallback object'})).toMatchSnapshot();
  });

  test('retrieve fallback object, when version does not match stored entry', () => {
    setItem('somekey', {
      foo: 'bar'
    }, 1);

    expect(getItem('somekey', 2, {some: 'fallback object'})).toMatchSnapshot();
  });

  test('retrieve fallback object, when there is no entry with given key', () => {
    expect(getItem('somekey', 2, {some: 'fallback object'})).toMatchSnapshot();
  });
});
