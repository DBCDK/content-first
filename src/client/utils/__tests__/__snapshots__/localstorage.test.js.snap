// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localstorage retrieve fallback object, when there is no entry with given key 1`] = `
Object {
  "some": "fallback object",
}
`;

exports[`localstorage retrieve fallback object, when version does not match stored entry 1`] = `
Object {
  "some": "fallback object",
}
`;

exports[`localstorage retrieves stored entry, when version matches 1`] = `
Object {
  "foo": "bar",
}
`;
